### PR TITLE
add collective pallet.

### PR DESCRIPTION
**Pull request by SaMotlagh**

### CODE REVIEW

**Comment by Rubberbandits** [comment]
Latest merge conflict resolution includes upping spec_version to 121

File: pallets/subtensor/Cargo.toml [hunk 1]
```diff
@@ -43,6 +46,8 @@ sp-tracing = { git = "https://github.com/paritytech/substrate", default-features
 parity-util-mem = { version = "0.11.0", features = ['primitive-types'] }
 rand = "0.8"
 sp-core = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.39" }
+# Used for sudo decentralization
+pallet-collective = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.39" }
```
Comment (shibshib) [inline pallets/subtensor/Cargo.toml]: Not sure if you need it in both test and production dependencies. You likely only need it in the main dependencies. If you need something for testing and not production then you can keep it only in dev dependencies, afaik.

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -320,6 +321,42 @@ impl pallet_sudo::Config for Runtime {
 	type RuntimeCall = RuntimeCall;
 }
 
+// Configure collective pallet for council
+parameter_types! {
+	pub const CouncilMotionDuration: BlockNumber = 3 * MINUTES;
+	pub const CouncilMaxProposals: u32 = 100;
+	pub const CouncilMaxMembers: u32 = 100;
```
Comment (shibshib) [inline runtime/src/lib.rs]: I don't think we need that many council members initially, or for testing, can we limit it to 3 for now?

**Review by shibshib — APPROVED** [review]

**Review by S0AndS0 — APPROVED** [review]

File: pallets/collective/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+[package]
+name = "pallet-collective"
+version = "4.0.0-dev"
+authors = ["Parity Technologies <admin@parity.io>"]
```
Comment (camfairchild) [inline pallets/collective/Cargo.toml]: Should be opentensor? Or at least adding opentensor

File: pallets/collective/src/migrations/v4.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (camfairchild) [inline pallets/collective/src/migrations/v4.rs]: Why do we need the migration? Does collective already exist in the chain storage?

**Review by camfairchild — COMMENTED** [review]
Review in progress, just had some questions 
@Rubberbandits

File: pallets/collective/Cargo.toml (hunk 3 above)
Comment (Rubberbandits) [reply pallets/collective/Cargo.toml]: Probably should add it, it’s partially modified

File: pallets/collective/src/migrations/v4.rs (hunk 4 above)
Comment (Rubberbandits) [reply pallets/collective/src/migrations/v4.rs]: It doesn’t, leftover from forking the pallet. I believe migrations are only triggered when specified essentially.

Comment (camfairchild) [reply pallets/collective/src/migrations/v4.rs]: > It doesn’t, leftover from forking the pallet. I believe migrations are only triggered when specified essentially.

Definitely should remove then

**Review by camfairchild — APPROVED** [review]

**Pull request merged**